diff --git a/gdk/gdkdisplaymanager.c b/gdk/gdkdisplaymanager.c
index e631652e0e..ae1cc1bbb6 100644
--- a/gdk/gdkdisplaymanager.c
+++ b/gdk/gdkdisplaymanager.c
@@ -298,8 +298,13 @@ gdk_display_manager_get (void)
 {
   static GdkDisplayManager *manager = NULL;
 
-  if (manager == NULL)
-    manager = g_object_new (GDK_TYPE_DISPLAY_MANAGER, NULL);
+  if (g_once_init_enter (&manager))
+    {
+      GdkDisplayManager *new_manager;
+
+      new_manager = g_object_new (GDK_TYPE_DISPLAY_MANAGER, NULL);
+      g_once_init_leave (&manager, new_manager);
+    }
   
   return manager;
 }
