diff --git a/meson.build b/meson.build
index 3a3e596..d5b05a0 100644
--- a/meson.build
+++ b/meson.build
@@ -43,7 +43,7 @@ libdrm = dependency(
 # We need to make sure we don't use any new libdrm functions, but those
 # are added very infrequently, so this is unlikely to be an issue.
-if libdrm.version().version_compare('<2.4.134')
+if libdrm.version().version_compare('<2.4.133')
   if libdrm.type_name() == 'internal'
     error('libdrm subproject out of date. Run `meson subprojects update`.')
   endif
 
